.ci/scripts/check_code_format.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -eEx
+set -o pipefail
+
+TMP_DIR="/tmp/$(basename "$0" .sh)_$$"
+rm -rf "${TMP_DIR}"
+mkdir -p "${TMP_DIR}"
+
+UCC_SRC_DIR="${TORCH_UCC_SRC_DIR}/ucc"
+cd "${UCC_SRC_DIR}"
+
+git log -1 HEAD
+git log -1 HEAD^
+echo "INFO: Checking code format on diff HEAD^..HEAD"
+#git-clang-format --binary=clang-format-11 --style=file --diff HEAD^ HEAD >"${TMP_DIR}/check_code_format.patch"
+git-clang-format --diff HEAD^ HEAD >"${TMP_DIR}/check_code_format.patch"
+echo "INFO: Generated patch file:"
+cat "${TMP_DIR}/check_code_format.patch"
+if [ "$(cat "${TMP_DIR}/check_code_format.patch")" = "no modified files to format" ]; then
+  echo "INFO: code format is OK"
+  exit 0
+fi
+
+git apply "${TMP_DIR}/check_code_format.patch"
+if ! git diff --quiet --exit-code; then
+  echo "ERROR: Code is not formatted according to the code style, see https://github.com/openucx/ucx/wiki/Code-style-checking for more info"
+fi
+
+rm -rf "${TMP_DIR}"
+exit 1
```

Comment:
exit 1 means test will fail and it will block PR merge button?

Reply:
Yes, if code format is not OK, this check failed and overall CI failed.

Reply:
last time we agreed that code formatting shouldn't be a blocker to merge PR https://github.com/openucx/ucc/pull/27

Reply:
Made code format check non-critical (0 exit code even if there're code format issues)